### ISSUE TITLE
Unexpected behavior for MLX array

### ISSUE THREAD

**Issue by sck-at-ucy** [ml-explore/mlx#176]
Am I doing something stupid here or is there a deeper issue? 

This works 

```
nx, ny = 5_000, 5_000  # Grid size
T = mx.zeros([nx, ny])  # Temperature array

T_mid = T[1:-1, 1:-1]
T[1:nx-1, 1:ny-1] = T_mid
```

while this raises an error 

```
nx, ny = 5_000, 5_000  # Grid size
T = mx.zeros([nx, ny])  # Temperature array

T_mid = T[1:-1, 1:-1]
T[1:-1, 1:-1] = T_mid
```

```
Traceback (most recent call last):
  File "/Users/m2/PycharmProjects/pythonProject_MLX/2D_Heat_Equation_MLX_Aproduction.py", line 41, in <module>
    T[1:-1, 1:-1] = T_mid
    ~^^^^^^^^^^^^
ValueError: Shapes (4998,4998) and (0,0) cannot be broadcast.

```

**Comment by awni** [comment]
Thanks for catching and filing the issue. The fix is in #189

**Comment by sck-at-ucy** [comment]
You were quick 👍🏻🙂
